transcom/mymove
# INT B-20809 GSR can add appeal to serious incident

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20809)

### NOTE
This PR also contains some UI compiling fixes that were displaying using USDWS parameters. Adjusting those to not use:
- `@include u-radius('05');`
- `@include u-minw(200px);`

fixed the compiling errors

- also contains efforts to fix some common flaky tests

## Summary

With the work [done in B-20808](https://github.com/transcom/mymove/pull/13994) where a GSR user can add appeals to violations, we also want to apply that functionality to serious incidents.

Since serious incidents occur at the evaluation report level and not the report violation level, we had to create a separate endpoint with the goal of being able to differentiate between the two when displaying move history (work not yet complete).

A lot of this work piggybacks off of the previous BL item, but I did do some refactoring in order to use the beauty of reusable components in React by adding in a couple new props into the `AppealModal` and dynamically rendering what the appeal is being added for into the modal (serious incident vs violation).

This PR does the following:
- adds new endpoint where an office user can add an appeal to a serious incident
- adds service object under `evaluationreport` package that handles adding an appeal to a serious incident
- refactors `ViolationAppealModal` to be `AppealModal` with added props for reusability
- refactors variables in the `EvaluationReportView` component for reusability/reduction of similar code
- adds UI tests
- adds server tests
- updates playwright test to cover GSR serious incident actions

### How to test

1. In your `.envrc`, turn the GSR flag to `true`
2. Log in as either a GSR user or a multi office user
3. You can create an evaluation report as a GSR user, so go ahead and create one and ensure you mark that there is a serious incident involved
4. You should now see the `Leave Appeal Decision` button when viewing the report
5. Leave an appeal decision on the serious incident (modal will pop up)
6. Check to make sure it populates in the database and displays on the UI
7. Ensure you can't leave another appeal on the serious incident (similar to violations)
8. Log in as another type of user (QAE for example)
9. View the previously created evaluation report
10. Ensure you can see the appeals
11. Create a new evaluation report as the other role with a serious incident
12. View the report, make sure you can't see the `Leave Appeal Decision` button
13. Log in as the GSR, pull up the report and confirm you CAN see the `Leave Appeal Decision` button

## Screenshots
<img width="746" alt="Screenshot 2024-11-08 at 10 52 33 AM" src="https://github.com/user-attachments/assets/f12cbbd8-955a-4fbb-9500-125ff9f75868">
<img width="1727" alt="Screenshot 2024-11-08 at 10 52 59 AM" src="https://github.com/user-attachments/assets/42a1c97f-74b3-4093-879c-5b16151f7f99">
<img width="503" alt="Screenshot 2024-11-08 at 10 53 11 AM" src="https://github.com/user-attachments/assets/1e0c55a5-4ae6-4d5e-9054-288b8c559902">
<img width="761" alt="Screenshot 2024-11-08 at 11 00 51 AM" src="https://github.com/user-attachments/assets/9559bf9a-afe2-4056-afab-488f728ce873">
<img width="1725" alt="Screenshot 2024-11-08 at 11 01 22 AM" src="https://github.com/user-attachments/assets/9ced4fbb-8b03-4807-973e-bdd92f1f44a4">
